### PR TITLE
Sorts: Some cleanup in data structures for representing sorts

### PR DESCRIPTION
**Pull request by blishko**
As a preparation for support for array sorts, this PR cleans up and (hopefully) simplifies the data structures  for Sort representation.

The next planned step is to add support for representing and cons-hashing sorts with arguments (e.g., array sorts).

### CODE REVIEW

**Comment by blishko** [comment]
This also fixes the memory leak in `Interpret::newSort` reported in #178.
Fixes #178.

**Comment by blishko** [comment]
With the current version, it is possible to support `Array` sorts once `Array` symbol is declared internally by `Logic`.

File: src/sorts/SSort.h [hunk 1]
```diff
@@ -28,214 +28,110 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
...
+    void free(SSymRef)
+    {
+        RegionAllocator<uint32_t>::free(SortSymbolWord32Size());
+    }
 };
 
-
 class SortAllocator : public RegionAllocator<uint32_t>
 {
     static sortid_t static_uniq_id;
```
[the remark this answers is not in the document]
Comment (blishko) [reply src/sorts/SSort.h]: Fixed.

File: src/sorts/SStore.h [hunk 2]
```diff
@@ -28,70 +28,59 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #define SSTORE_H
 
 #include "SSort.h"
-#include "StringMap.h"
 #include "Alloc.h"
+#include "TypeUtils.h"
 
+
+#include <unordered_map>
 #include <iosfwd>
```
[the remark this answers is not in the document]
Comment (blishko) [reply src/sorts/SStore.h:36]: `std::ostream` is used as a parameter in this header, so I would keep this.

File: src/sorts/SStore.h [hunk 3]
```diff
@@ -28,70 +28,59 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
...
+        return true;
     }
-    IdRef newIdentifier(const char* name, vec<int>& nl) {
-        IdStrRef nr = isa.alloc(name);
-        return ia.alloc(nr, nl);
+};
+
+struct SortHash {
+    uint32_t operator () (const SortKey& s) const {
+        uint32_t v = (uint32_t)s.sym.x;
```
[the remark this answers is not in the document]
Comment (blishko) [reply src/sorts/SStore.h]: Fixed.

File: src/sorts/SStore.h [hunk 4]
```diff
@@ -28,70 +28,59 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
...
+        for (int i = 0; i < s.args.size(); i++) {
```
[the remark this answers is not in the document]
Comment (blishko) [reply src/sorts/SStore.h]: Fixed.

**Review by aehyvari — APPROVED** [review]
Looks good now!

**Pull request merged**